src/ast.rs

```diff
@@ -393,8 +419,14 @@ impl From<Literal> for RawExpression {
     }
 }
 
-impl From<Expression> for RawExpression {
-    fn from(x: Expression) -> Self {
+impl From<Identifier> for RawExpression {
+    fn from(x: Identifier) -> Self {
+        RawExpression::Variable(x)
```

Comment:
This _could_ be fine, but I have concern bells going off in my head. In the C++ parser, this would resolve to a NameRef (expr? statement?), the ideas being A) then we only need store a given identifier string 1x, and B) could there be non-variable identifiers in this context (e.g., function names in invocations)?

This could also just be a symptom of incremental development (which is a good thing!). If you agree w/my concerns, then think about putting a comment here...although we'll probably remember, if it comes to it.

Reply:
Oh also, a param isn't a "Variable", per se. /shrug

Reply:
I also disliked the name Variable being used for function arguments. Changed:
`Binding(Identifier)`

Reply:
> This could be fine, but I have concern bells going off in my head. In the C++ parser, this would resolve to a NameRef (expr? statement?), the ideas being A) then we only need store a given identifier string 1x, and B) could there be non-variable identifiers in this context (e.g., function names in invocations)?

Sorry, I don't really understand the problem. Can you explain F2F?

Reply:
A) Not an issue for now; we're going to have `Identifiers` hold their associated strings.

B) I think this is resolved in my head by renaming it `Binding`.

So...lgtm.